### DISCUSSION TITLE
S7 communication error with Giesen

### DISCUSSION THREAD

**Discussion by rossmcintyre-greenfarm** [artisan-roaster-scope/artisan#1070] [Q&A] [answered]
Hi

I am hoping someone can help me, I have connected my Giesen to artisan successfully but after creating a new button the following error message comes up after activating it:
S7 Communication Error writeInt: type set doesn't define_round_method@line 581

Would someone be able to shed some light on what I have missed? 

The new button carries out the correct command on the Artisan screen correctly but it does not adjust the roaster

Thanks in advance

Ross

**Comment by MAKOMO** [comment]
Could you please share the button definition (you can attach a screen shot here via drag-and-drop to a comment box on Githubs Web) as well as the exact error message (maybe also via a screenshot!?).

Please indicate also which Artisan version and build you are using.

**Comment by rossmcintyre-greenfarm** [comment, marked as answer]
![Screenshot 2023-01-10 085806](https://user-images.githubusercontent.com/122296823/211507916-72488b31-5d0d-4abb-8c92-25b2b88a12e8.png)
![Screenshot 2023-01-10 085918](https://user-images.githubusercontent.com/122296823/211507957-4568ab00-8826-4e95-82d7-fadda6b73664.png)
Here are both the screenshots of the error and the button. I have artisan 2.8.0

Thanks!

**Reply by poundy** [reply]
which is the new button??

**Reply by MAKOMO** [reply]
The command is wrong. It should read

```
setDBint(2,14,{})
```
without that `11`. The placeholder `{}` is replaced on activation by the calculated value resulting from subtracting `10? from  the current burner setting.

**Reply by rossmcintyre-greenfarm** [reply]
> The command is wrong. It should read
> 
> ```
> setDBint(2,14,{})
> ```
> 
> without that `11`. The placeholder `{}` is replaced on activation by the calculated value resulting from subtracting `10? from the current burner setting.

Understood, that has worked

Thank you very much for your help!
